library/src/scala/tasty/Reflection.scala

```diff
@@ -2409,21 +2409,25 @@ trait Reflection { reflection =>
   // IMPLICIT SEARCH //
   /////////////////////
 
-  // TODO: this should not be top level
-  /** Find an implicit of type `T` in the current scope given by `ctx`.
-   *  Return an `ImplicitSearchResult`.
-   *
-   *  @param tpe type of the implicit parameter
-   *  @param ctx current context
-   */
-  def searchImplicit(tpe: Type): ImplicitSearchResult
+  /** Result of an implicit search */
+  type ImplicitSearch <: AnyRef
+
+  given TypeTest[ImplicitSearch, ImplicitSearchSuccess] = ImplicitSearchSuccessTypeTest
+  protected val ImplicitSearchSuccessTypeTest: TypeTest[ImplicitSearch, ImplicitSearchSuccess]
 
-  type ImplicitSearchResult <: AnyRef
+  val ImplicitSearch: ImplicitSearchModule
 
-  given TypeTest[ImplicitSearchResult, ImplicitSearchSuccess] = ImplicitSearchSuccessTypeTest
-  protected val ImplicitSearchSuccessTypeTest: TypeTest[ImplicitSearchResult, ImplicitSearchSuccess]
+  trait ImplicitSearchModule { self: ImplicitSearch.type =>
+    /** Find an implicit of type `T` in the current scope given by `ctx`.
```

Comment:
```suggestion
    /** Find a given instance of type `T` in the current scope given by `ctx`.
```

Reply:
Maybe `Summon a given instance of type ...`

Reply:
I would say that the compiler summons a value by looking at the given instances.